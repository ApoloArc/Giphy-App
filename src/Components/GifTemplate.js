import React from 'react'
import PropTypes from 'prop-types'
export const GifTemplate = ({title, url})=> {
    return (
        <div className='card animate__animated animate__fadeInDown'>
            <img src={url} alt={title}/>
            <p className='title-card'>{title}</p>
        </div>
    )
}

GifTemplate.propTypes={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
