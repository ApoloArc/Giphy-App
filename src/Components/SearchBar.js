import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const SearchBar = ({setCategories}) => {

    const [busqueda, setBusqueda] = useState('');
    
    const cambioBusqueda = (e)=>{
        setBusqueda(e.target.value);
    }

    const sinRefresh = e =>{
        e.preventDefault();
        
        setCategories(e=>[busqueda, ...e]);
        setBusqueda('');
    }

    return (
        <form onSubmit={sinRefresh}>
            <input
            type='text'
            placeholder='Busca algo'
            value={busqueda}
            onChange={cambioBusqueda}
            >
            </input>
        </form>
    )
}

SearchBar.propTypes= {
    setCategories: PropTypes.func.isRequired,
}