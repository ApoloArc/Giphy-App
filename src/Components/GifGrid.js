import { GifTemplate } from './GifTemplate';
import { useFetchGifs } from './Hooks/useFetchsGifs';
import PropTypes from 'prop-types';


const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category);

    return (
        <div>
            <h3 className='titulo-category'>{category}</h3>
            {loading && <p className='animate__animated animate__fadeInDown'>Cargando, espera compare...</p>}
            <div className='grid'>
            {
               images.map(img => <GifTemplate key={img.id} {...img}/>)
            }
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
