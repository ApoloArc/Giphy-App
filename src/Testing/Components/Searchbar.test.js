import {shallow} from 'enzyme'
import { SearchBar } from '../../Components/SearchBar';

describe('Probando el SearchBar', () => {
    

    const setCategories = jest.fn();
        
    let wrapper = shallow(<SearchBar setCategories={setCategories}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<SearchBar setCategories={setCategories}/>);
    })

    test('Aprobar la funcion SetCategory', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de funcionar el input', () => {
        
        const input = wrapper.find('input');

        const value = 'Hola giles';

        input.simulate('change', {target: {value}})
    
    })
    
    test('No debe ser llamado el input ni una vez', () => {
        
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(setCategories).not.toHaveBeenCalled();

    })
    
    test('debe de llamar el SetCategories y limpiar la caja', () => {
        
        const input = wrapper.find('input');

        const value = 'Hola Mundo';       
        input.simulate('change', {target: {value}})
        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
})
