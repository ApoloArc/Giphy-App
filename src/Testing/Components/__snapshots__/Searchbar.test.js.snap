// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el SearchBar Aprobar la funcion SetCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Busca algo"
    type="text"
    value=""
  />
</form>
`;
